lib/replicate.js

```diff
@@ -35,8 +35,27 @@ function replicate (feed, initiator, opts) {
         stream.close(feed.discoveryKey)
         return
       }
+      if (opts.onfeedauthenticate) {
+        if (!stream.remotePublicKey) {
+          stream.once('handshake', function () {
```

Comment:
prob have to setMaxListeners(0) here to avoid leak warnings if we use the event for flow control.

Reply:
I was wondering about this: `.setMaxListeners` on stream is _weird_ in this location, shouldn't it be in the `Protocol` constructor of `hypercore-protocol` as the stream may be passed-in from the outside. ([here](https://github.com/hypercore-protocol/hypercore-protocol/blob/c299117fbcbe7100adf442f30768ec4860cacda7/index.js#L358) maybe?) Anyways I will add it.

Reply:
I normally avoid events for flow control in general because of this, but unsure if you can do this without it. Since this is the one taking on the responsibility of adding that listener, I'd put it here, so it's clear that we aren't just suppressing a leak from the outside.